lib/kafo_parsers/puppet_strings_module_parser.rb

```diff
@@ -125,7 +129,11 @@ def sanitize(value)
     end
 
     def merge_doc_types(types)
-      Hash[@parsed_hash['signatures'].flatten.map { |param| [ param['name'], param['type'] ] }].merge(types)
+      Hash[tag_params.select { |param| !param['types'].nil? }.map { |param| [ param['name'], param['types'].first ] }].merge(types)
+    end
+
+    def tag_params
+      @parsed_hash['docstring']['tags'].select { |tag| tag['tag_name'] == 'param' }
```

Comment:
Thanks, good catch. I handled nil in `tag_params`, fixed another bug in handling 'defaults' and added a test for a class without parameters.